src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/AncientAltarTask.java

```diff
@@ -140,10 +146,14 @@ private void abort() {
 
     private void finish() {
         if (running) {
-            dropLocation.getWorld().playSound(dropLocation, Sound.ENTITY_ZOMBIE_VILLAGER_CURE, 1F, 1F);
-            dropLocation.getWorld().playEffect(dropLocation, Effect.STEP_SOUND, Material.EMERALD_BLOCK);
-            dropLocation.getWorld().dropItemNaturally(dropLocation.add(0, -0.5, 0), output);
 
+            AncientAltarOutputEvent ancientAltarOutputEvent = new AncientAltarOutputEvent(output, altar, player);
+            Bukkit.getPluginManager().callEvent(ancientAltarOutputEvent);
+            if (!ancientAltarOutputEvent.isCancelled()) {
+                dropLocation.getWorld().playSound(dropLocation, Sound.ENTITY_ZOMBIE_VILLAGER_CURE, 1F, 1F);
+                dropLocation.getWorld().playEffect(dropLocation, Effect.STEP_SOUND, Material.EMERALD_BLOCK);
+                dropLocation.getWorld().dropItemNaturally(dropLocation.add(0, -0.5, 0), output);
```

Comment:
Since you are able to modify the output of the item you should also use the output item from that event rather than the original one.